src/cobra/core/species.py

```diff
@@ -55,8 +57,52 @@ def reactions(self) -> FrozenSet:
         FrozenSet
             A frozenset that includes the reactions of the species.
         """
+        if self.model and self.__class__.__name__ == "Gene":
+            return frozenset(self.model.reactions.query(lambda x: self in x, "genes"))
+        elif self.model and self.__class__.__name__ == "Metabolite":
+            return frozenset(
+                self.model.reactions.query(lambda x: self in x, "metabolites")
+            )
         return frozenset(self._reaction)
 
+    def reaction_add(
+        self, reaction: "Reaction", context: Optional[HistoryManager] = None
+    ) -> None:
+        """Add reaction to .reaction field, with context.
+
+        If this is called with a context, will be reversed when exiting the context.
+
+        Parmeters
+        ---------
+        reaction: cobra.Reaction
+        context: HistoryManager, optional
+            context this action is in, optional (defualt None).
+        """
+        self._reaction.add(reaction)
+        if context:
```

Comment:
I would not provide the context as an argument. You can just get the context with `get_context(reaction._model)`.